ci/Jenkinsfile

```diff
@@ -68,18 +71,33 @@ pipeline {
       artifactNumToKeepStr: '3',
     ))
   }
-
   stages {
-    stage('Download') {
+     stage('Download') {
       steps {
         script {
-          sh "mkdir '${env.WORKSPACE}/tmp/'"
-        }
-        fileOperations([
-          fileDownloadOperation(url:"${params.BUILD}", userName:"${CI_CREDS_USR}", password:"${CI_CREDS_PSW}", targetLocation:"./tmp", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
-        ])
-        script {
-          sh "tar -zxvf '${env.WORKSPACE}/tmp/StatusIm-Desktop.tar.gz' -C '${env.WORKSPACE}/tmp'"
+          /* Creating directory for application build */
+          sh "mkdir -p '${env.WORKSPACE}/tmp/pkg/'"
+          def job_name = "${JOB_NAME}"
+          /* Searching the latest build of the master branch is available only for ci.infra.status.im */
+          if(job_name.endsWith("nightly")){
+            echo "Searching the latest successful build for the master branch."
+            if(params.BUILD != ""){
+              echo "BUILD value was ignored. Please use 'ci.status.in' to test specific builds."
+            }
+            /* Seraching and downloading the latest successful build for the master branch */
+            copyArtifacts(projectName: "status-desktop/systems/linux/x86_64/package/", filter: 'pkg/*-x86_64.tar.gz', selector: lastWithArtifacts(), target: "${env.WORKSPACE}/tmp")
+          } else {
+            if(params.BUILD == ""){
+              throw new Exception("BUILD is required field.")
+            }
+            withCredentials([usernamePassword(credentialsId: 'status-im-auto', usernameVariable: 'CI_CREDS_USR', passwordVariable: 'CI_CREDS_PSW')]) {
+              fileOperations([
+                fileDownloadOperation(url:"${params.BUILD}", userName:"${env.CI_CREDS_USR}", password:"${env.CI_CREDS_PSW}", targetLocation:"./tmp/pkg/", targetFileName:"StatusIm-Desktop.tar.gz", proxyHost:"", proxyPort:"")
```

Comment:
If you insist on supporting file download, it should not use any auth, and it should use publicly available build URLs.

And it should be in a separate stage using `when` clauses.

Reply:
Credentials removed